tests/tests_integration/test_api/test_assets.py

```diff
@@ -54,9 +55,59 @@ def post_spy(cognite_client):
         yield
 
 
+@pytest.fixture(scope="session")
+def twenty_assets(cognite_client: CogniteClient) -> AssetList:
+    prefix = "twenty_assets:"
+    root = AssetWrite(external_id=f"{prefix}root", name="root")
+    assets = AssetWriteList(
+        [root]
+        + [
+            AssetWrite(
+                external_id=f"{prefix}asset{i}",
+                name=f"asset{i}",
+                parent_external_id=root.external_id,
+            )
+            for i in range(19)
+        ]
+    )
+    return cognite_client.assets.upsert(assets, mode="replace")
+
+
 @pytest.fixture(scope="module")
-def root_test_asset(cognite_client):
-    return cognite_client.assets.retrieve(external_id="test__asset_0")
+def root_test_asset(cognite_client) -> Asset:
+    root = AssetWrite(external_id="test__asset_root", name="test__asset_root")
+    retrieved = cognite_client.assets.retrieve(external_id=root.external_id)
+    if retrieved is not None:
+        return retrieved
+    hierarchy = generate_asset_tree(root, first_level_size=5, size=1000, depth=8)
+    cognite_client.assets.create_hierarchy(hierarchy, upsert=True, upsert_mode="replace")
+    return cognite_client.assets.retrieve(external_id=root.external_id)
+
+
+def generate_asset_tree(root: AssetWrite, first_level_size: int, size: int, depth: int) -> list[AssetWrite]:
+    np.random.seed(0)
+    # A power law distribution describes the real shape of an asset hierarchy, i.e., few roots, many leaves.
+    count_per_level = np.random.power(0.2, depth)
+    count_per_level.sort()
+    total = count_per_level.sum()
+    count_per_level = (count_per_level / total) * size
+    count_per_level = np.ceil(count_per_level).astype(np.int64)
+    count_per_level[0] = first_level_size
+    count_per_level[-1] = size - count_per_level[:-1].sum() - 1
+    last_level = [root]
+    hierarchy = [root]
+    for level, count in enumerate(count_per_level, 1):
+        this_level = []
+        for asset_no in range(count):
+            parent = random.choice(last_level)
```

Comment:
You only set the seed for numpy, so the use of built-in `random` makes this function non-deterministic

Reply:
We have a context manager that handles all this btw